EuchreSolution/Euchre/GameScript/animation.ts

```diff
@@ -26,72 +21,122 @@ function makeCardElem(cardID: string, flippedUp: boolean): HTMLDivElement {
 	return card;
 }
 
-function animMoveCard(cardID: string, top: string, left: string, z?: string): void {
-	const div = document.getElementById(cardID) as HTMLDivElement | null;
-	if (!div) { return; }
+function animMoveCard(cardOrIdOrElement: Card | string | HTMLDivElement,
+	top: string, left: string, z?: string): void {
 
-	div.style.top = top;
-	div.style.left = left;
+	const cardElement = getCardElement(cardOrIdOrElement);
+	if (!cardElement) { return; }
+
+	cardElement.style.top = top;
+	cardElement.style.left = left;
 	if (z) {
-		div.style.zIndex = z;
+		cardElement.style.zIndex = z;
 	} else {
-		div.style.zIndex = zIndex.toString();
+		cardElement.style.zIndex = zIndex.toString();
+		zIndex++;
 	}
-	zIndex++;
 }
 
-function animDeal(hands: Card[][], trumpCandidate: Card, dealer: Player, settings: Settings): void {
-	if (!controller || controller.isStatMode()) { return; }
+function animDeal(hands: Card[][], trumpCandidate: Card, dealer: Player,
+	settings: Settings, callback: () => void): void {
+
+	if (!controller || controller.isStatMode()) {
+		callback();
+		return;
+	}
 
-	let player: Player;
-	let delay: number; //delay to second round deal
-	let cardID: string;
-	let flippedUp: boolean;
-	let cardElem: HTMLElement | null;
 	const isOpenHands = settings.openHands;
 	const hasHooman = settings.hasHooman;
 
-	animClearTable();
-
-	player = getNextPlayer(dealer);
-	delay = 0;
-
-	makeCardElem("deck", false);
-	makeCardElem(trumpCandidate.id, false);
+	const trumpCandidateId = trumpCandidate.id;
+	const startDealDelegate = () => {
+		animClearTable();
+		animPlaceDealerButt(dealer);
+		makeCardElem("deck");
+		makeCardElem(trumpCandidateId);
+		for (const hand of hands) {
+			for (const card of hand) {
+				makeCardElem(card.id);
+			}
+		}
+	};
+	AnimController.queueAnimation(AnimType.DealHands, startDealDelegate);
 
+	let player = dealer;
 	for (let i = 0; i < hands.length; i++) {
```

Comment:
This loop for the first and second rounds are almost identical. Can we factor them out?

Reply:
Done